salt/_states/metalk8s.py

```diff
@@ -41,44 +52,96 @@ def static_pod_managed(name,
+        return _error(ret, "Source must be a single string")
+
     if not config_files:
         config_files = []
 
     for config_file in config_files_opt or []:
         if __salt__["file.file_exists"](config_file):
             config_files.append(config_file)
+        else:
+            log.debug(
+                "Ignoring optional config file %s: file does not exist",
+                config_file
+            )
+
+    config_file_digests = []
+    for config_file in config_files:
+        try:
+            digest = __salt__["hashutil.digest_file"](
+                config_file, checksum="sha256"
+            )
+        except CommandExecutionError as exc:
+            return _error(
+                ret,
+                "Unable to compute digest of config file {}: {}".format(
+                    config_file, exc
+                )
+            )
+        config_file_digests.append(digest)
 
-    config_file_digests = [
-        __salt__["hashutil.digest_file"](config_file, checksum="sha256")
-        for config_file in config_files
-    ]
     config_digest = __salt__["hashutil.md5_digest"](
         "-".join(config_file_digests)
     )
 
     match = re.search(r'metalk8s-(?P<version>.+)$', __env__)
     metalk8s_version = match.group('version') if match else "unknown"
 
-    return __states__["file.managed"](
-        name,
-        source,
-        template=kwargs.pop("template", "jinja"),
-        user=kwargs.pop("user", "root"),
-        group=kwargs.pop("group", "root"),
-        mode=kwargs.pop("mode", "0600"),
-        makedirs=kwargs.pop("makedirs", False),
-        backup=kwargs.pop("backup", False),
-        context=dict(
-            context or {},
-            config_digest=config_digest, metalk8s_version=metalk8s_version
-        ),
-        **kwargs
+    context_ = dict(
+        context or {},
+        config_digest=config_digest, metalk8s_version=metalk8s_version
     )
 
+    if __opts__["test"]:
+        log.warning("Test functionality is not yet implemented.")
+        ret["comment"] = (
+            "The manifest {} is in the correct state (supposedly)."
+        ).format(name)
+        return ret
+
+    # Gather the source file from the server
+    try:
+        source_filename, source_sum, comment_ = __salt__["file.get_managed"](
+            name,
+            template="jinja",
+            source=source,
+            source_hash="",
+            source_hash_name=None,
+            user="root",
+            group="root",
+            mode="0600",
+            attrs=None,
+            saltenv=__env__,
+            context=context_,
+            defaults=None,
+        )
+    except Exception as exc:  # pylint: disable=broad-except
+        log.debug(traceback.format_exc())
+        return _error(ret, "Unable to get managed file: {}".format(exc))
+
+    if comment_:
+        return _error(ret, comment_)
+    else:
+        try:
+            return __salt__["metalk8s.manage_static_pod_manifest"](
```

Comment:
Thinking of it, shouldn't this be called `static_pod_manifest_managed` similar to `file.managed`?

Reply:
This module is a "copy" of salt `file.manage_file` so for me the name is good here (especially since it's an execution module and not a state

Reply:
Argh good point!

Reply:
I can rename this state function `static_pod_manifest_managed` (from `static_pod_managed`) if you want, but this line is about the execution function (which mirrors `file.manage_file` in principle).